server/jamovi/server/dataset/datacache.py

```diff
@@ -83,3 +91,39 @@ def get_value(self, row: int, column: int) -> CellValue:
             self._cache[cell_range] = values
         value = values[row - row_start][column - column_start]
         return value
+    
+
+class DataWriteBuffer:
+    
+    _write_buffer: dict[CellCoordinate, CellValue]
+    _set_values: Callable[[dict[CellCoordinate, CellValue]], None]
+    # Note: index_name is included to mimic the get 
+    _max_items: int
+
+    def __init__(
+            self, 
+            set_values_method: Callable[[dict[CellCoordinate, CellValue]], None],
+            max_items: int = 50
+            ):
+        self._write_buffer = {}
+        self._set_values = set_values_method
+        self._max_items = max_items
+
+    def stage(self, cell_coordinate: CellCoordinate, value: CellValue):
+        """Stage a value for writing to the database"""
+        self._write_buffer[cell_coordinate] = value
+        if len(self._write_buffer) >= self._max_items:
+            self.commit()
+
+    # commit the buffer
+    def commit(self):
+        """Write the contents of the buffer to the database using the set_values(...) 
+        method injected to the constructor"""
+
+        self._set_values(self._write_buffer)
+
+        self._write_buffer.clear()
+    
+    def dump(self):
```

Comment:
`clear()`, `discard()`? `dump()` has a different meaning thanks to `json.dump()`

having said that, i can't think of a situation where we'd want to discard the cached values.